spdxexp/satisfies.go

```diff
@@ -50,6 +50,26 @@ func Satisfies(testExpression string, allowedList []string) (bool, error) {
 	return false, nil
 }
 
+// ExtractLicenses extracts licenses from the given expression without duplicates.
+// Returns an array of licenses or error if error occurs during processing.
+func ExtractLicenses(expression string) ([]string, error) {
+	node, err := parse(expression)
+	if err != nil {
+		return nil, err
+	}
+
+	expanded := node.expand(true)
+	licenses := make([]string, 0)
+	allLicenses := flatten(expanded)
+	for _, license := range allLicenses {
+		licenses = append(licenses, license.lic.license)
```

Comment:
Are there circumstances under which `license.lic` is `nil`? It seems like that's _possible_, but I don't know if it actually happens. Wouldn't want a panic in this code.

Reply:
Check for nilness added!